traits/trait_types.py

```diff
@@ -2774,6 +2774,169 @@ def items_event(self):
+        """ Get the mapped value for a value. """
+        return self.map[value]
+
+    def post_setattr(self, object, name, value):
+        setattr(object, name + "_", self.mapped_value(value))
+
+    def info(self):
+        keys = sorted(repr(x) for x in self.map.keys())
+        return " or ".join(keys)
+
+    def get_editor(self, trait):
+        from traitsui.api import EnumEditor
+
+        return EnumEditor(values=self, cols=trait.cols or 3)
+
+
+class PrefixMap(Map):
+    """ A cross between the PrefixList and Map classes.
+
+    Like Map, PrefixMap is created using a dictionary, but in this
+    case, the keys of the dictionary must be strings. Like PrefixList,
+    a string *v* is a valid value for the trait attribute if it is a prefix of
+    one and only one key *k* in the dictionary. The actual values assigned to
+    the trait attribute is *k*, and its corresponding mapped attribute is
+    *map*[*k*].
+
+    Example
+    -------
+    ::
+
+        mapping = {'true': 1, 'yes': 1, 'false': 0, 'no': 0 }
+        boolean_map = PrefixMap(mapping)
+
+    This example defines a Boolean trait that accepts any prefix of 'true',
+    'yes', 'false', or 'no', and maps them to 1 or 0.
+
+    Parameters
+    ----------
+    map : dict
+        A dictionary whose keys are strings that are valid values for the
+        trait attribute, and whose corresponding values are the values for
+        the shadow trait attribute.
+
+    Attributes
+    ----------
+    map : dict
+        A dictionary whose keys are strings that are valid values for the
+        trait attribute, and whose corresponding values are the values for
+        the shadow trait attribute.
+    """
+
+    def __init__(self, map, **metadata):
+        self.map = map
+        self._map = _map = {}
+        for key in map.keys():
+            _map[key] = key
+        self.fast_validate = (ValidateTrait.prefix_map, _map, self.validate)
+
+        default_value = metadata.pop("default_value", None)
+
+        if default_value is not None:
+            try:
+                default_value = self.validate(None, None, default_value)
+            except TraitError:
+                raise ValueError("The default value for PrefixMap trait must "
+                                 "be a unique prefix of a key in the map.")
+        super(Map, self).__init__(default_value, **metadata)
+
+    def validate(self, object, name, value):
+        try:
+            if value not in self._map:
+                match = None
+                n = len(value)
+                for key in self.map.keys():
+                    if value == key[:n]:
+                        if match is not None:
+                            match = None
+                            break
+                        match = key
+                if match is None:
+                    self.error(object, name, value)
+                self._map[value] = match
+            return self._map[value]
+        except:
```

Comment:
There's another bare `except` here to address. (In general, if a PR reviewer notices a particular issue in a PR, it's worth checking the rest of the PR for similar issues.)

Here it would also be good to reduce the scope of the `try / except` if possible.